File: src/main/java/net/volcano/jdacommands/model/command/annotations/argument/Arg.java
package net.volcano.jdacommands.model.command.annotations.argument;

import org.intellij.lang.annotations.Language;

import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

@Target({ElementType.PARAMETER})
@Retention(RetentionPolicy.RUNTIME)
public @interface Arg {
	
	/**
	 * Min value for numerical input
	 */
	double min() default Double.MIN_VALUE;
	
	/**
	 * Max value for numerical input
	 */
	double max() default Double.MAX_VALUE;
	
	boolean defaultToCaller() default false;
	
	/**
	 * Regex for matcher argument
	 */
	@Language("RegExp")
	String regex() default "";
	
	/**
	 * Regex flags
	 */
	int flags() default 0;
	
	/**
	 * If roles need to be from the same guild as the command is issued from
	 */
	boolean sameGuild() default true;
	
	String usage() default "DEFAULT";
	
}

File: src/main/java/net/volcano/jdacommands/model/command/annotations/argument/Arg.kt
package net.volcano.jdacommands.model.command.annotations.argument

@Target(AnnotationTarget.VALUE_PARAMETER)
@Retention(AnnotationRetention.RUNTIME)
annotation class Arg(

	/**
	 * Min value for numerical input
	 */
	val min: Long = Long.MIN_VALUE,

	/**
	 * Max value for numerical input
	 */
	val max: Long = Long.MAX_VALUE, val defaultToCaller: Boolean = false,

	/**
	 * Regex for matcher argument
	 */
	val regex: String = "",

	/**
	 * Regex flags
	 */
	val flags: Int = 0,

	/**
	 * If roles need to be from the same guild as the command is issued from
	 */
	val sameGuild: Boolean = true, val usage: String = "DEFAULT"

)